set.mm

```diff
@@ -517517,24 +517671,290 @@ commutative monoid (=vectors) together with a semiring (=scalars) and a
         VUIVUJYPVUKYPYAVUQUAFVURVULVUMVUNXM $.
     $}
 
-    $d .x. f i v $.  $d .x. g h w $.  $d A f i v $.  $d A g h w $.
-    $d B f i v $.  $d F f i v $.  $d F g h $.  $d M f i v $.  $d M g h w $.
-    $d R f i v $.  $d R g h $.  $d X g $.  $d f i ph v $.  $d h i v w $.
-    elrgspn.1 $e |- ( ph -> X e. B ) $.
+    $d .x. f g h i v w y $.  $d A f g h i v w $.  $d B f g i v w y $.
+    $d F f g h i v w y $.  $d M f g h i v w y $.  $d R f g h i v w y $.
+    $d X f i g v $.  $d ph f g i v w y $.
     $( Membership in the subring generated by the subset ` A ` .  An element
        ` X ` lies in that subring if and only if ` X ` is a linear combination
        with integer coefficients of products of elements of ` A ` .
        (Contributed by Thierry Arnoux, 5-Oct-2025.) $)
     elrgspn $p |- ( ph -> ( X e. ( N ` A ) <-> E. g e. F
        X = ( R gsum ( w e. Word A |-> ( ( g ` w ) .x. ( M gsum w ) ) ) ) ) ) $=
```

Comment:
Why is the new proof so much longer? Only to eliminate the former hypothesis "( ph -> X e. B )"?

Reply:
Yes, only to eliminate the hypothesis.
I eliminated the hypothesis by proving both `X e. B` from both sides of the final biconditional, which requires closure of the sums, both the (infinite) larger sum and the embedded sum over each word.